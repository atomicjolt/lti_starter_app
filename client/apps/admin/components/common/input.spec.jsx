import React from 'react';
import TestRenderer from 'react-test-renderer';
import Input from './input';

describe('input', () => {
  let result;
  let props;
  let changed;
  let instance;

  beforeEach(() => {
    changed = false;
    props = {
      inputProps: {
        id: 'IM AN ID',
        value: 'IM A VALUE',
        checked: true,
        name: 'the name',
        type: 'radio',
        onChange: () => { changed = true; },
      },
      className: 'imaclass',
      labelText: 'IMA LABEL',
    };
    result = TestRenderer.create(<Input {...props} />);
    instance = result.root;
  });

  it('matches the snapshot', () => {
    expect(result).toMatchSnapshot();
  });

  it('handles the onChange function', () => {
    expect(changed).toBeFalsy();
    instance.findByType('input').simulate('change');
    expect(changed).toBeTruthy();
  });
});
