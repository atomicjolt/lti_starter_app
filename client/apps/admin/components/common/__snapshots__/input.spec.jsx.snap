// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`input matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Input
    className="imaclass"
    inputProps={
      Object {
        "checked": true,
        "id": "IM AN ID",
        "name": "the name",
        "onChange": [Function],
        "type": "radio",
        "value": "IM A VALUE",
      }
    }
    labelText="IMA LABEL"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <label
          className="imaclass"
          htmlFor="IM AN ID"
        >
          <input
            checked={true}
            id="IM AN ID"
            name="the name"
            onChange={[Function]}
            type="radio"
            value="IM A VALUE"
          />
          <span>
            IMA LABEL
          </span>
        </label>,
        undefined,
      ],
      "className": "c-input--container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            undefined,
            <input
              checked={true}
              id="IM AN ID"
              name="the name"
              onChange={[Function]}
              type="radio"
              value="IM A VALUE"
            />,
            <span>
              IMA LABEL
            </span>,
          ],
          "className": "imaclass",
          "htmlFor": "IM AN ID",
        },
        "ref": null,
        "rendered": Array [
          undefined,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "checked": true,
              "id": "IM AN ID",
              "name": "the name",
              "onChange": [Function],
              "type": "radio",
              "value": "IM A VALUE",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "IMA LABEL",
            },
            "ref": null,
            "rendered": "IMA LABEL",
            "type": "span",
          },
        ],
        "type": "label",
      },
      undefined,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <label
            className="imaclass"
            htmlFor="IM AN ID"
          >
            <input
              checked={true}
              id="IM AN ID"
              name="the name"
              onChange={[Function]}
              type="radio"
              value="IM A VALUE"
            />
            <span>
              IMA LABEL
            </span>
          </label>,
          undefined,
        ],
        "className": "c-input--container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              undefined,
              <input
                checked={true}
                id="IM AN ID"
                name="the name"
                onChange={[Function]}
                type="radio"
                value="IM A VALUE"
              />,
              <span>
                IMA LABEL
              </span>,
            ],
            "className": "imaclass",
            "htmlFor": "IM AN ID",
          },
          "ref": null,
          "rendered": Array [
            undefined,
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "checked": true,
                "id": "IM AN ID",
                "name": "the name",
                "onChange": [Function],
                "type": "radio",
                "value": "IM A VALUE",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "IMA LABEL",
              },
              "ref": null,
              "rendered": "IMA LABEL",
              "type": "span",
            },
          ],
          "type": "label",
        },
        undefined,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
